.github/workflows/linux_meson.yml

```diff
@@ -71,7 +71,7 @@ jobs:
       run: |
         python -m pip install git+https://github.com/numpy/numpy.git
         python -m pip install ninja cython pytest pybind11 pytest-xdist pytest-timeout click rich-click doit pydevtool pooch hypothesis "setuptools<67.3"
-        python -m pip install git+https://github.com/serge-sans-paille/pythran.git@e3babfd
+        python -m pip install git+https://github.com/serge-sans-paille/pythran.git
```

Comment:
Why don't we switch back to released versions here, now that 0.15 is out?

Reply:
Do we need to bump `pythran` to `>=0.15` in `pyproject.toml` too?

Reply:
Not just yet, it's only been out for a few days and there are CI failures involving Pythran right now.

Reply:
Also, aside from some setuptools interaction that I don't know in detail, pythran 0.14 is actually plenty for building scipy.

Reply:
@h-vetinari don't we want to keep this as is in this 'install from repositories' job? `pythran` has been done like this for at least 2 years, see https://github.com/scipy/scipy/blame/8d846f19ac89a2940faa6a356044dc12acc89ac7/.github/workflows/linux_meson.yml.

Reply:
Fair point. Out of the two occurrences of `.../pythran.git` in https://github.com/scipy/scipy/commit/d1da380d8e357df1d5f8e24558f334c5593eca92, one got already removed, and the other one was pre-existing. As long as we removed the work-arounds introduced between pythran 0.14 & 0.15, I'm fine with leaving things as-is.